test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -1078,6 +1078,57 @@ ConstructorTestClass(int i, bool b)
+            $utilityCommandNounsWithConvertToVerbSingleQuote = $allUtilityCommandNounsSingleQuote | Where-Object { $_ -in "'CliXml'", "'Csv'", "'Html'", "'Json'", "'Xml'" }
+            $utilityCommandNounsWithConvertToVerbDoubleQuote = $allUtilityCommandNounsDoubleQuote | Where-Object { $_ -in """CliXml""", """Csv""", """Html""", """Json""", """Xml""" }
+            $utilityCommandNounsWithConvertToVerbStartingWithC = $allUtilityCommandNouns | Where-Object { $_ -in 'CliXml', 'Csv' }
+            $utilityCommandNounsWithConvertToVerbStartingWithCSingleQuote = $allUtilityCommandNounsSingleQuote | Where-Object { $_ -in "'CliXml'", "'Csv'" }
+            $utilityCommandNounsWithConvertToVerbStartingWithCDoubleQuote = $allUtilityCommandNounsDoubleQuote | Where-Object { $_ -in """CliXml""", """Csv""" }
+        }
+
+        It "Should complete Noun for '<TextInput>'" -TestCases @(
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Noun "; ExpectedNouns = $allUtilityCommandNouns }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Noun '"; ExpectedNouns = $allUtilityCommandNounsSingleQuote }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Noun """; ExpectedNouns = $allUtilityCommandNounsDoubleQuote }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Noun F"; ExpectedNouns = $utilityCommandNounsStartingWithF  }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Noun 'F"; ExpectedNouns = $utilityCommandNounsStartingWithFSingleQuote }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Noun ""F"; ExpectedNouns = $utilityCommandNounsStartingWithFDoubleQuote }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Verb ConvertTo -Noun "; ExpectedNouns = $utilityCommandNounsWithConvertToVerb  }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Verb ConvertTo -Noun '"; ExpectedNouns = $utilityCommandNounsWithConvertToVerbSingleQuote }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Verb ConvertTo -Noun """; ExpectedNouns = $utilityCommandNounsWithConvertToVerbDoubleQuote }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Verb ConvertTo -Noun C"; ExpectedNouns = $utilityCommandNounsWithConvertToVerbStartingWithC  }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Verb ConvertTo -Noun 'C"; ExpectedNouns = $utilityCommandNounsWithConvertToVerbStartingWithCSingleQuote }
+            @{ TextInput = "Get-Command -Module $utilityModuleName -Verb ConvertTo -Noun ""C"; ExpectedNouns = $utilityCommandNounsWithConvertToVerbStartingWithCDoubleQuote }
+        ) {
+            param($TextInput, $ExpectedNouns)
+            $res = TabExpansion2 -inputScript $TextInput -cursorColumn $TextInput.Length
+            $completionText = $res.CompletionMatches.CompletionText
+            $completionText -join ' ' | Should -BeExactly ($ExpectedNouns -join ' ')
```

Comment:
```suggestion
            $completionText -join ' ' | Should -Be ($ExpectedNouns -join ' ')
```

Reply:
So perhaps for linux we just do comparison which should be case insenstive. Or we do lower case comparison.

Reply:
I believe Get-Command is doing different sorting for each, although this also a typo when we released ConvertFrom-CliXml/ConvertTo-CliXml cmdlets. 

They should have both been lowercased like Export-CliXml/Import-Clixml. 

But it does bring up interesting behaviour with sorting between Linux and Windows.

Reply:
> $nouns = (Get-Command -Module $module).Noun | Sort-Object -Unique

This returns `Xml`, not both.

Reply:
> $nouns = (Get-Command -Module $module).Noun | Sort-Object -Unique

With -CaseSensitive this will return both `Clixml` and `CliXml`.

Reply:
Ah good point. That should be an easy fix then.

Reply:
Although this means we need to remove `OrdinalIgnoreCase` for SortedSet.

Reply:
Also completion result does not give back case sensitive completions. it seems to have its own logic or something which is preventing case sensitive completions to come back.

I think workaround here is to use another method.

Reply:
E.g. GetMatchingResults will give both `CliXml` and `Clixml`, but `CompleteArgument will emit `CliXml`.

Reply:
And above they match if we use `-CaseSensitive`, but this does not help since completions returns only one, and this is where It gets hard to guarantee which one shows up depending on OS.